network/ssh.go

```diff
@@ -59,11 +64,45 @@ func NewSSHAgent(dialer Dialer) (*SSHAgent, error) {
 		return nil, err
 	}
 
-	return &SSHAgent{
-		Agent:  keyring,
-		Dialer: dialer,
-		User:   defaultUser,
-	}, nil
+	sockDir, err := ioutil.TempDir("", "mantle-ssh-")
+	if err != nil {
+		return nil, err
+	}
+
+	// Use a similar naming scheme to ssh-agent
+	sockPath := fmt.Sprintf("%s/agent.%d", sockDir, os.Getpid())
+	sockAddr := &net.UnixAddr{Name: sockPath, Net: "unix"}
+	listener, err := net.ListenUnix("unix", sockAddr)
+	if err != nil {
+		os.RemoveAll(sockDir)
+		return nil, err
+	}
+
+	a := &SSHAgent{
+		Agent:    keyring,
+		Dialer:   dialer,
+		User:     defaultUser,
+		Socket:   sockPath,
+		sockDir:  sockDir,
+		listener: listener,
+	}
+
+	go func() {
+		for {
+			conn, err := listener.Accept()
```

Comment:
As far as I know yes, it appeared to be a common go pattern but I don't personally know the details (yet)